flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ExecutionAttemptMappingProvider.java

```diff
@@ -38,32 +38,34 @@
     private final List<ExecutionVertex> tasks;
 
     /** The cached mapping, which would only be updated on miss. */
-    private final LinkedHashMap<ExecutionAttemptID, ExecutionVertex> cachedTasksById;
+    private volatile Map<ExecutionAttemptID, ExecutionVertex> cachedTasksById;
 
     public ExecutionAttemptMappingProvider(Iterable<ExecutionVertex> tasksIterable) {
         this.tasks = new ArrayList<>();
         tasksIterable.forEach(this.tasks::add);
 
-        this.cachedTasksById =
-                new LinkedHashMap<ExecutionAttemptID, ExecutionVertex>(tasks.size()) {
-
-                    @Override
-                    protected boolean removeEldestEntry(
-                            Map.Entry<ExecutionAttemptID, ExecutionVertex> eldest) {
-                        return size() > tasks.size();
-                    }
-                };
+        this.cachedTasksById = new ConcurrentHashMap<>(tasks.size());
```

Comment:
That's too much of synchronization imo.

Reply:
Yes, it looks like our map is immutable so I can use HashMap here